util/stopper.go

```diff
@@ -162,13 +162,20 @@ func (s *Stopper) SetStopped() {
 	}
 }
 
-// Quiesce moves the stopper to state draining, waits until all tasks
-// complete, then moves back to non-draining state. This is used from
+// Quiesce moves the stopper to state draining. Unquiesce() needs to
+// be called before the stopper is stopped. This is used from
 // unittests.
 func (s *Stopper) Quiesce() {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	s.draining = 1
+	atomic.StoreInt32(&s.draining, 1)
+}
```

Comment:
I don't think you should remove the call to `s.drain.Wait()` here. The name `Quiesce()` suggests that the tasks should in fact drain by the time it returns.